requirement/src/main/java/org/arquillian/cube/requirement/Requirements.java

```diff
@@ -26,4 +33,24 @@ public static void checkRequirement(Requires requires, Annotation context) throw
             }
         }
     }
+
+    private static Requirement getRequirement(Annotation context) {
+        Requirement requirement = null;
+
+        final ServiceLoader<Requirement> requirementLoader = ServiceLoader.load(Requirement.class);
+
+        final Iterator<Requirement> requirements = requirementLoader.iterator();
+
+        while (requirement == null && requirements.hasNext()) {
+            Requirement req = requirements.next();
+            try {
+                req.getClass().getDeclaredMethod("check", context.annotationType());
```

Comment:
I struggle to understand how this code is supposed to be working. We never really `check` if given implementation satisfies the requirement, but always just return the last passing this loop. And only then we call the actual `.check` in the caller method `checkRequirement`.  Will this always work?

Reply:
@bartoszmajsak , We are looking for requirement with respective method definition `check(context.annotationType)`. If it haven't found it it will throw exception by saying `NoSuchMethodException` which we are catching & iterating over next. Here we are returning first requirement with correct method(`check(RequiresDocker)|check(RequiresKubernetes)`) on classpath, we are not returning last passing value. This should always work

Reply:
I this given case, yes. But what if you have two SPIs with the same method signature (unlikely, we are not preventing it from happening)?

Reply:
And I still believe we are returning the last matching one, not the first one.

Reply:
> Here we are returning first requirement with correct method

Right, but the main concern I explained above still remains.

Reply:
yeah but this is the limitation of spi. May be we can add weight to take precedence of one over another wdyt?

Reply:
That's not a limitation of SPI really, more this code to be honest. But on the other we don't need it yet. Worth documenting it though.